tests/python_package_test/test_arrow.py

```diff
@@ -103,6 +101,29 @@ def test_dataset_construct_fuzzy(
     assert filecmp.cmp(tmp_path / "arrow.txt", tmp_path / "pandas.txt")
 
 
+# -------------------------------------------- FIELDS ------------------------------------------- #
+
+
+@pytest.mark.parametrize("field", ["label", "weight"])
+def test_dataset_construct_fields_fuzzy(field: str):
+    arrow_table = generate_random_arrow_table(3, 1000, 42)
+    arrow_array = generate_random_arrow_array(1000, 42)
+
+    arrow_dataset = lgb.Dataset(arrow_table, **{field: arrow_array})
+    arrow_dataset.construct()
+
+    pandas_dataset = lgb.Dataset(arrow_table.to_pandas(), **{field: arrow_array.to_numpy()})
+    pandas_dataset.construct()
+
+    np_assert_array_equal(arrow_dataset.get_field(field), pandas_dataset.get_field(field))
+    np_assert_array_equal(
+        getattr(arrow_dataset, f"get_{field}")(), getattr(pandas_dataset, f"get_{field}")()
+    )
```

Comment:
Instead of `pytest.parametrize()`-ing over field names and doing stuff like this, could you please just write a deep test that just explicitly invokes the methods being tested?

Like this one:

https://github.com/microsoft/LightGBM/blob/aeafccfbfb5c223d33b61ebe0f1e8b5592249151/tests/python_package_test/test_basic.py#L502

I appreciate the effort to limit repetition here, but I think `getattr()` + an f-string to retrieve a method and then calling that is more indirection and opportunity for silent issues than I'm comfortable with in unit test code. I'm willing to accept a bit more indirection in exchange for these tests being a bit more accessible to contributors who are not as familiar with Python.

Reply:
Fair point, this felt quite hacky :smile: